### DISCUSSION TITLE
Upgrading Netbox from v2.7.11 to the latest stable release.

### DISCUSSION THREAD

**Discussion by aquelf** [netbox-community/netbox#9667] [Q&A]
Hi team, I have recently been tasked with upgrading our Netbox version from v2.7.11 to the latest stable Netbox v3 release.
I understand that I may need to upgrade to an intermediary version & from there to the latest.

Can any one point me in the right direction as to what the intermediary release should be, before running the upgrade script? 

The server that the app is hosted on is OS = CentOS 7 (64-bit) is the process the same for CentOS as it is for Ubuntu?

I'm also awaiting to get root access to the server, in the meantime I was wondering is there a sure way of identifying whether the Netbox was cloned from the repository or installed from a tar.gz file. 

Any help would be greatly appreciated, thanks in advance.

**Comment by candlerb** [comment]
v2.11.12 is the intermediate version you should go through. See https://github.com/netbox-community/netbox/discussions/8396#discussioncomment-2006622 for a step-by-step guide. As per that guide, I strongly suggest you build a fresh v2.11.12 instance and migrate data into that, rather than attempting an in-place upgrade.

Beware that Netbox 3.2 requires newer versions of dependencies, especially Python 3.8 and Postgres 10.  Unless you're really wedded to CentOS 7 and are happy to go hunting for EPEL or other third-party packages, maybe this is time to rebuild on a new platform. CentOS 7 is [end of life](https://endoflife.software/operating-systems/linux/centos) in 2 years anyway.

>  in the meantime I was wondering is there a sure way of identifying whether the Netbox was cloned from the repository or installed from a tar.gz file.

If directory `/opt/netbox/.git` exists, then it was cloned.  You can also confirm using:

```
cd /opt/netbox
git status
```

**Reply by aquelf** [reply]
Great thanks for the great explanation & guidance, ideally we would like to built the install on Ubuntu however we aren't in a position to spin up a Ubuntu machine or CentOs Stream as we cannot go past CentOS 7.6  at the moment.

Does CentOS 7.6 support Python 3.8 & Postgres, or I guess my question is, can I install v2.11.12 on a new CentOS 7.6 & will it support the new features on Netbox 3.x.x? Thanks again.

**Reply by candlerb** [reply]
Not out-of-the-box: http://mirror.centos.org/centos/7/os/x86_64/Packages/ has python3.6.8 and postgresql 9.2.24.

As I said before, it's up to you to find newer packages which will run on this platform.

**Comment by mtinberg** [comment]
The python38 runtime is part of SCL (Software Collections, newer versions of common software than what shipped with the base OS)

yum install centos-release-scl centos-release-scl-rh on centos to enable these repos, then yum install rh-python38 which will put it in /opt/rh/.  yum search rh-python38 to find all the other pre-built modules for setuputils or pip or whatever, I listed below what I have installed.  You can use the scl command to make it the default for your shell, it works much like enabling a virtualenv, or in scripts but mostly this isn't necessary as for netbox you need to just create a virtualenv.

/opt/rh/rh-python38/root/bin/python3.8 -m venv venv

any issues I ran into I fixed in my fork of Ansible Role lae.netbox https://github.com/mtinberg/ansible-role-netbox/tree/netbox-3.0 if I've forgotten to mention them.  One thing I did do was both install the centos7 package for uwsgi, just to get the file paths and permissions, and also the current version inside the netbox virtualenv, which is what is actually used for the service.

$ rpm -qa | grep rh-python
rh-python38-runtime-2.0-4.el7.x86_64
rh-python38-python-pip-19.3.1-3.el7.noarch
rh-python38-python-setuptools-wheel-41.6.0-7.el7.noarch
rh-python38-python-setuptools-41.6.0-7.el7.noarch
rh-python38-2.0-4.el7.x86_64
rh-python38-python-3.8.13-1.el7.x86_64
rh-python38-python-libs-3.8.13-1.el7.x86_64
rh-python38-python-psycopg2-2.8.4-5.el7.x86_64
rh-python38-python-rpm-macros-3.8.13-1.el7.noarch
rh-python38-python-devel-3.8.13-1.el7.x86_64
rh-python38-python-pip-wheel-19.3.1-3.el7.noarch

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Reply by aquelf** [reply]
Cheers, thanks for the help guys. I was still having issues with this.  Is there any achieved guides on how to install older versions of Netbox, I need to try get up a test instance of 2.7.11 or 2.11.12 but based on the updated docx & trying to use some CentOS 7 recommended installs, I keep having issues with Python, Django, postgres etc. Thanks

**Reply by candlerb** [reply]
If you can show exactly what you did, and what specific errors you see, you may be able to get help.

Otherwise, I am not aware of a specific document "how to install Netbox 2.11 on CentOS 7".

**Comment by mtinberg** [comment]
You can install the SCL of Python38 and that works fine on RHEL7/CentOS7, I'm doing that with Netbox 3.2.x and the lae.netbox Ansible playbook (with some modifications to support SCL python38 in /opt/rh instead of the default system-wide one).  Highly suggest building another venv for each version though as Brian said, so one for 2.11.12 and one for 3.2.5, leaving the 2.7.11 instance alone, getting them configured, backing up the db with pg_dump then running the upgrade.sh for 2.11.12 then for 3.2.5 to get the db up-to-date.  PostgreSQL also have the latest version (13.x IIRC) built for RHEL7 in their own yum repo, and there may be an SCL version of PostgreSQL that meets the minimum requirement but it's probably going to work better long-term to run the latest PG than the minimum.  You'll still need the RH packaged client libraries though, as that is what he RH packaged python pg db lib will be built against.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
